crates/astria-sequencer/src/sequencer.rs

```diff
@@ -98,3 +104,44 @@ impl Sequencer {
         Ok(())
     }
 }
+
+fn start_grpc_server(storage: &cnidarium::Storage, grpc_addr: std::net::SocketAddr) -> Result<()> {
+    use ibc_proto::ibc::core::{
+        channel::v1::query_server::QueryServer as ChannelQueryServer,
+        client::v1::query_server::QueryServer as ClientQueryServer,
+        connection::v1::query_server::QueryServer as ConnectionQueryServer,
+    };
+    use penumbra_tower_trace::remote_addr;
+    use tonic::transport::Server;
+    use tower_http::cors::CorsLayer;
+
+    let ibc = penumbra_ibc::component::rpc::IbcQuery::new(storage.clone());
+    let cors_layer = CorsLayer::permissive();
+
+    // TODO: setup HTTPS?
+    let grpc_server = Server::builder()
+        .trace_fn(|req| {
+            if let Some(remote_addr) = remote_addr(req) {
+                tracing::error_span!("grpc", ?remote_addr)
+            } else {
+                tracing::error_span!("grpc")
+            }
+        })
+        // (from Penumbra) Allow HTTP/1, which will be used by grpc-web connections.
+        // This is particularly important when running locally, as gRPC
+        // typically uses HTTP/2, which requires HTTPS. Accepting HTTP/2
+        // allows local applications such as web browsers to talk to pd.
+        .accept_http1(true)
+        // (from Penumbra) Add permissive CORS headers, so pd's gRPC services are accessible
+        // from arbitrary web contexts, including from localhost.
+        .layer(cors_layer)
+        .add_service(ClientQueryServer::new(ibc.clone()))
+        .add_service(ChannelQueryServer::new(ibc.clone()))
+        .add_service(ConnectionQueryServer::new(ibc.clone()));
+
+    tokio::task::Builder::new()
```

Comment:
This creates a handle to a future that itself is a tokio handle, which is a bit much. Just make this method an `async fn start(...) -> Result<(), Error>` and assign this to a variable in the `run_until_stopped` so that you can join your futures/tasks at the same time.

Also, it's not ideal (because we cannot attach our own span to the spawned task), but it's better to use this

However, in this case it's fine to use https://docs.rs/tonic/latest/tonic/transport/server/struct.Router.html#method.serve_with_shutdown